pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2877,6 +2877,23 @@ type CephFilesystemSubVolumeGroupSpec struct {
 	// list of Ceph Filesystem volumes with `ceph fs volume ls`. To learn more about Ceph Filesystem
 	// abstractions see https://docs.ceph.com/en/latest/cephfs/fs-volumes/#fs-volumes-and-subvolumes
 	FilesystemName string `json:"filesystemName"`
+	// Pinning configuration of CephFilesystemSubVolumeGroup,
+	// reference https://docs.ceph.com/en/quincy/cephfs/fs-volumes/#pinning-subvolumes-and-subvolume-groups
+	// +optional
+	Pinning CephFilesystemSubVolumeGroupSpecPinning `json:"pinning,omitempty"`
+}
+
+// CephFilesystemSubVolumeGroupSpecPinning represents the pinning configuration of SubVolumeGroup
+type CephFilesystemSubVolumeGroupSpecPinning struct {
+	// Pinning type of CephFilesystemSubVolumeGroup
+	// +optional
+	// +kubebuilder:default=distributed
```

Comment:
How does this default work during upgrades? Will an upgraded cluster get the default of 1? (where this setting was not available)

Reply:
As they can get `distributed` and `1` unless they plan to disable it

Reply:
Instead of doing a type, value version where the type and value are different data types, why don't we create a typed struct?
https://go.dev/play/p/SxlYzUyVayO

Reply:
Thanks updated the changes